File: typescrip-bun/integration/example.test.ts
import { test, expect, beforeAll, describe } from "bun:test";
import { connection, dag } from "@dagger.io/client";


describe("Example", () => {
  test(
    "containerEcho",
    async () => {
        let res!: string;

        await connection(async () => {
          res = await dag.example().containerEcho("test").stdout();
        });

        expect(res).toBe("test\n");
    },
    { timeout: 30 * 1000 }
  );
});

File: go/integration/example_test.go
package integration

import (
	"context"
	"testing"

	"dagger/example/integration/client/dag"

	"github.com/stretchr/testify/require"
)

func TestContainerEcho(t *testing.T) {
	ctx := context.Background()
	res, err := dag.Example().ContainerEcho("test").Stdout(ctx)
	require.NoError(t, err)
	require.Equal(t, res, "test\n")
}
